cmd/kube-rbac-proxy/app/kube-rbac-proxy.go

```diff
@@ -277,14 +264,17 @@ func createKubeRBACProxyConfig(opts *completedProxyRunOptions) (*server.KubeRBAC
 	return proxyConfig, nil
 }
 
+type runnerFunc func() error
+type interrupterFunc func(error)
```

Comment:
To be honest I considered that, but I still think that keeping it in the same file as the rest is probably clean enough, because:

- it's not code that is/will be used elsewhere within the project, as it's very specific to starting up the KRP
- it's a short piece of code (total 42 loc: typedefs, `secureServerRunner()` and `addToSafeWaitGroup()`), and mainly wrapper code; at the same time the file itself isn't very large (335 loc currently)
- based on the current structure, I'm not sure whether there's a better place, as there are not a lot of dirs within `cmd/kube-rbac-proxy/` and based on the first point it's not very fitting for `pkg/` IMHO

We could always refactor at a later point when we feel that the file is growing too much.

As a side note, this could be simplified a bit more if we dropped the typedefs for the two funcs; they just offer a tiny bit of clarity, but not much since everything is used around in the same spots. What do you think?

Reply:
Let it leave as is... but 335 loc is not long? 😄 Maybe it is me and I am odd, but I love small files where I can jump in between, rather than scrolling throw long texts.

Reply:
Well, based on what I've seen so far in our projects I'd say 335 might even be below average :sweat_smile:

I have a suggestion if we want to reduce loc further: we can go with what you've suggested and split runner/interrupter to another file, and we can also split `completedProxyRunOptions` and `Complete()` to another file too. This will cut it down to approx. 200 loc :slightly_smiling_face: WDYT?

Reply:
hahaha, yeah. In general in the cloud-space, 3000 LOC seems to be the magical threshold and not 300 😛 

Lets keep it as is. We might want to revisit the whole logic at some point at all.